resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/localstore/FileSystemResourceManager.java

```diff
@@ -1286,6 +1286,16 @@ private void prepareWrite(IFile target, IFileInfo fileInfo, int updateFlags, boo
 				}
 			}
 		}
+		if (BitMask.isSet(updateFlags, IResource.REPLACE)) {
+			// forced update of derived flag during IFile.write:
+			Resource resource = (Resource) target;
+			ResourceInfo info = resource.getResourceInfo(false, false);
+			if (BitMask.isSet(updateFlags, IResource.DERIVED)) {
+				info.set(ICoreConstants.M_DERIVED);
```

Comment:
This could probably be even happening without REPLACE.

Reply:
If we remove the REPLACE condition we would set derived=false on every normal setContent(FLAG). That is not what we want.

Reply:
I was speaking of this particular case (setting DERIVED to true) doesn't require REPLACE. Here REPLACE would only be useful in order to un-set derived.
But as mentioned earlier, the case of changing value and derived at the same time seems not so useful and not so trivial to implement properly. We could probably just ditch it by restraining how `derived` is used in `write`

Reply:
ok, i think i implemented that now  se below.